Scripts/script-DocumentationAutomation.yml

```diff
@@ -9,30 +9,96 @@ script: |-
   reload(sys)
   sys.setdefaultencoding('utf-8')
 
-  entryID = demisto.args()['entryID']
+  def get_yaml_obj(entry_id):
+      yml_file_path = demisto.getFilePath(entry_id)['path']
+      with open(yml_file_path, 'r') as yml_file:
+          data = yaml.safe_load(yml_file)
 
-  res = demisto.executeCommand('getFilePath', {'id': entryID})
-  if isError(res[0]):
-      demisto.results(res)
-      sys.exit(0)
+      return data
 
-  path = res[0]['Contents']['path']
 
-  yamlFile  = open(path)
-  dataMap = yaml.safe_load(yamlFile)
-  yamlFile.close()
+  def get_command_examples(entry_id):
+      if entry_id is None:
+          return []
+
+      examples_path = demisto.getFilePath(entry_id)['path']
+      with open(examples_path, 'r') as examples_file:
+          data = examples_file.read().split('\n')
+
+      return data
+
+
+  def find_cmd_example(command_examples, cmd):
+      valid_examples = [example for example in command_examples
+          if example.startswith(cmd) or example.startswith('!{}'.format(cmd))]
+
+      return valid_examples[0] if len(valid_examples) > 0 else ' '
+
+
+  def extract_command(cmd_example):
+      parts = cmd_example.split(' ')
+
+      return parts[0], dict([p.replace('"', '').split('=') for p in parts[1:]])
+
+
+  def generate_command_example(cmd, cmd_example):
+      context_example = ''
+      md_example = ''
+      if cmd_example != ' ':
+          try:
+              example_cmd, kwargs = extract_command(cmd_example)
+              res = demisto.executeCommand(example_cmd, kwargs)
+
+              for entry in res:
+                  if isError(entry):
+                      demisto.results(res)
+                      raise RuntimeError('something went wrong with your command.')
+
+              raw_context = entry.get('EntryContext')
+              context = {k.split('(')[0] : v for k, v in raw_context.items()}
+
+              context_example = json.dumps(context, indent=4)
+              md_example = entry.get('HumanReadable', json.dumps(entry['Contents']))
+
+          except RuntimeError as e:
+              demisto.results("The provided example for cmd {} has failed...".format(cmd['name']))
+              demisto.results(res)
+          except Exception as e:
+              demisto.results('Error encountered in the processing of command {}, error was: {}'.format(cmd['name'], str(e)) +
+                  ". Please check your command inputs and outputs")
+      else:
+          demisto.results('did not get any example for {}. please add it manually.'.format(cmd['name']))
```

Comment:
this needs to be -:
* triggered only if `commands` file is given.
* collected and shown as a single entry.